Lib/typing.py

```diff
@@ -998,7 +1005,7 @@ def get_type_hints(obj, globalns=None, localns=None):
                 if value is None:
                     value = type(None)
                 if isinstance(value, str):
-                    value = ForwardRef(value)
+                    value = ForwardRef(value, is_argument=True)
```

Comment:
Sorry for not having time to review this earlier, this all looks good except for the meaning is exactly the opposite to what I expected :-) There are two options:
* Rename `is_argument` to `is_not_argument` everywhere (arguably ugly)
* Carefully replace `True` with `False` and vice versa, and update the corresponding checks

@nnja @ambv Will any of you have time to make a PR?

Reply:
@ilevkivskyi cc/ @ambv Yes, I'll have time tomorrow evening on my flight home.

Reply:
Ivan, I don't think the meaning is reversed? As I understand it:
* "argument" is what's directly an annotation on a function argument and so on; so it includes non-types
* otherwise, it has to be a type (like inside generics)

If you still want to make a change to this, I would prefer renaming to `is_annotation` which signifies that we're talking about an entire annotations.

Reply:
Renaming to `is_annotation` is a possible option, but IMO is still less clear than `is_argument` with reversed value, since the latter makes simple distinction between these two cases:
* `List['int']`, here `'int'` appears as a type argument in another type
* `'int'` here `'int'` appears just as a type
In the first case the rules are stricter, e.g. `ClassVar` can't appear as a type argument.

In view of this, the _ideal_ option would be to _both_ rename it to `is_type_argument` and reverse `False` to `True`, but I don't want to ask for too much :-) I will be happy with either option if there will be a short comment explaining what `is_argument`/`is_annotation`/`is_type_argument` means.

Reply:
Is the term "type argument" used anywhere in typing? I only heard of them being called "generics".

Reply:
This is the term used (among other synonyms) in PEP 484 and co and in mypy. In `typing` they are all just `__args__`.

Reply:
OK, cool, in this case you're right, let's just reverse True with False and vice versa.

Reply:
@ambv @ilevkivskyi Should I open a new PR for these changes? What's the needed course of action for the backport? https://github.com/python/cpython/pull/6912

Reply:
Yes, please make a PR against master branch. Also it makes sense to do this ASAP, since 3.7rc1 is out in one day.

Reply:
@ilevkivskyi 👍I can work on this tonight. Will you be available to review?

Reply:
@ilevkivskyi Done! The updated PR is https://github.com/python/cpython/pull/7039. Please let me know how it looks.